Source/SwiftLintFramework/Rules/TrailingWhitespaceRule.swift

```diff
@@ -9,25 +9,34 @@
 import Foundation
 import SourceKittenFramework
 
-public struct TrailingWhitespaceRule: CorrectableRule, ConfigurationProviderRule,
-                                      SourceKitFreeRule {
+public struct TrailingWhitespaceRule: CorrectableRule, ConfigurationProviderRule {
 
-    public var configuration = TrailingWhitespaceConfiguration(ignoresEmptyLines: false)
+    public var configuration = TrailingWhitespaceConfiguration(ignoresEmptyLines: false,
+                                                               ignoresComments: true)
 
     public init() {}
 
     public static let description = RuleDescription(
         identifier: "trailing_whitespace",
         name: "Trailing Whitespace",
         description: "Lines should not have trailing whitespace.",
-        nonTriggeringExamples: [ "//\n" ],
-        triggeringExamples: [ "// \n" ],
-        corrections: [ "// \n": "//\n" ]
+        nonTriggeringExamples: [ "let name: String\n", "//\n", "// \n",
+            "let name: String //\n", "let name: String // \n" ],
+        triggeringExamples: [ "let name: String \n", "/* */ let name: String \n" ],
+        corrections: [ "let name: String \n": "let name: String\n",
+            "/* */ let name: String \n": "/* */ let name: String\n"]
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
         let filteredLines = file.lines.filter {
-            $0.content.hasTrailingWhitespace() &&
+            let commentKinds = SyntaxKind.commentKinds()
+            if $0.content.hasTrailingWhitespace() && configuration.ignoresComments,
+                let lastSyntaxKind = file.syntaxKindsByLines[$0.index].last
+                where commentKinds.contains(lastSyntaxKind) {
+                return false
+            }
+
+            return $0.content.hasTrailingWhitespace() &&
```

Comment:
On non violating case, `$0.content.hasTrailingWhitespace()` is evaluated twice. The closure should return `false` first if `$0.content.hasTrailingWhitespace()` returns `false`.